module/icp/algs/aes/aes_impl.c

```diff
@@ -22,12 +22,13 @@
  * Copyright (c) 2003, 2010, Oracle and/or its affiliates. All rights reserved.
  */
 
-#if defined(_KERNEL) && defined(__amd64)
+#if defined(__amd64)
 #include <linux/simd_x86.h>
+#endif
 
+#if defined(_KERNEL) && defined(__amd64)
 #define	KPREEMPT_DISABLE	kfpu_begin()
 #define	KPREEMPT_ENABLE		kfpu_end()
-
 #else
 #define	KPREEMPT_DISABLE
 #define	KPREEMPT_ENABLE
```

Comment:
The `linux/simd_x86.h` header already takes care of defining `kfpu_begin/end` properly for the build environment.  You should be able to simply unconditionally define `KPREEMPT_DISABLE` to be `kfpu_begin()`.  I wouldn't be at all against a search and replace of `KPREEMPT_DISABLE` with `kfpu_begin()`, etc.  Even if it's a slight divergence from Illumos.